typescript/rebalance_bot/contracts.ts

```diff
@@ -0,0 +1,98 @@
+import { ethers } from "ethers";
+import { BotConfig } from "../../config/types";
+
+// Typed contract interfaces for better type safety
+export interface DLoopCoreContract extends ethers.Contract {
```

Comment:
Excellent improvement to type safety! These interfaces provide proper typing for all contract methods, eliminating the need for unsafe `any` casts. This will catch type errors at compile time rather than runtime.